central/detection/alertmanager/alert_manager_impl.go

```diff
@@ -229,12 +237,59 @@ func mergeNetworkFlowViolations(old, new *storage.Alert) bool {
 	return mergeAlertsByLatestFirst(old, new, storage.Alert_Violation_NETWORK_FLOW)
 }
 
+func mergeFileAccessViolations(oldAlert, newAlert *storage.Alert) bool {
+	newViolations := newAlert.GetFileAccessViolation()
+	oldViolations := oldAlert.GetFileAccessViolation()
+
+	numNewAccesses := len(newViolations.GetAccesses())
+	numOldAccesses := len(oldViolations.GetAccesses())
+
+	if numNewAccesses == 0 || numOldAccesses >= maxRunTimeViolationsPerAlert {
+		return false
+	}
+
+	// start with old violations and merge with new
+	mergedFileAccesses := oldViolations.GetAccesses()
+	lastAccessTime, err := lastFileTime(mergedFileAccesses)
+	if err != nil {
+		log.Errorf(
+			"Failed to merge alerts. "+
+				"New alert %s (policy=%s) has %d file accesses and old alert %s (policy=%s) has %d file accesses: %v",
+			newAlert.GetId(), newAlert.GetPolicy().GetName(), numNewAccesses,
+			oldAlert.GetId(), oldAlert.GetPolicy().GetName(), numOldAccesses, err,
+		)
+		// At this point, we know that the new alert has non-zero file violations but it cannot be merged.
```

Comment:
Put a comment above the function declaration with an explanation as to what this function does and returns.